utilities/utils.py

```diff
@@ -127,7 +131,7 @@ def train_validate_test_normal(
 
 
 def train(loader, model, opt, output_dim):
-    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```

Comment:
Should we add another condition where we do want to use multiple GPU or just wait until we use distributed parallelism?

Reply:
There are two types of data parallelism. 

DistributedDataParallelism (newer) -> multi-processing
DataParallelism (older) -> multi-threading

I would say that DistributedDataParallelism is more compelling to take advantage of distributed resources to perform the biggest chunk of the parallelization. 
DataParallelism could be combined with DistributedDataParallelism, but I would not make that our priority.

Reply:
To clarify my own understanding: DDP is distributed memory ("MPI" on one or more nodes), DP is only capable of single node, but still "MPI" parallel